src/dialogs/quick-open/ha-quick-open-dialog.ts

```diff
@@ -0,0 +1,216 @@
+import "@polymer/paper-input/paper-input";
+import "@material/mwc-list/mwc-list-item";
+import "@material/mwc-list/mwc-list";
+import {
+  css,
+  customElement,
+  html,
+  internalProperty,
+  LitElement,
+  property,
+} from "lit-element";
+import { fireEvent } from "../../common/dom/fire_event";
+import "../../components/ha-dialog";
+import { haStyleDialog } from "../../resources/styles";
+import { HomeAssistant } from "../../types";
+import { PolymerChangedEvent } from "../../polymer-types";
+import { fuzzySequentialMatch } from "../../common/string/sequence_matching";
+import { componentsWithService } from "../../common/config/components_with_service";
+import { domainIcon } from "../../common/entity/domain_icon";
+import { computeDomain } from "../../common/entity/compute_domain";
+import { domainToName } from "../../data/integration";
+
+export interface QuickOpenDialogParams {
+  entityFilter?: string;
+  commandMode?: boolean;
+}
+
+interface QuickOpenItem {
+  text: string;
+  domain: string;
+  icon?: string;
+  onClick(ev: Event): void;
+}
+
+@customElement("ha-quick-open-dialog")
+export class QuickOpenDialog extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @internalProperty() private _items: QuickOpenItem[] = [];
+
+  @internalProperty() private _entityFilter = "";
+
+  @internalProperty() private _opened = false;
+
+  @internalProperty() private _commandMode = false;
+
+  public async showDialog(params: QuickOpenDialogParams) {
+    this._commandMode = params.commandMode || false;
+    this._opened = true;
+  }
+
+  public closeDialog() {
+    this._opened = false;
+    this._entityFilter = "";
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  private _entityFilterChanged(ev: PolymerChangedEvent<string>) {
+    this._entityFilter = ev.detail.value;
+    if (this._entityFilter.startsWith(">")) {
+      this._commandMode = true;
+      this._entityFilter = this._entityFilter.substring(1);
+    } else {
+      this._commandMode = false;
+    }
+
+    let items: QuickOpenItem[];
+    if (this._commandMode) {
+      items = this._generateCommandItems(this._entityFilter);
+    } else {
+      items = this._generateEntityItems(this.hass, this._entityFilter);
+    }
+    this._items = this._filterItems(items, this._entityFilter);
+  }
+
+  private _generateCommandItems(_filter: string): QuickOpenItem[] {
+    const reloadableDomains = componentsWithService(this.hass, "reload").sort();
+
+    return reloadableDomains.map((domain) => ({
+      text:
+        this.hass.localize(`ui.dialogs.quick_open.commands.${domain}`) ||
+        this.hass.localize(
+          "ui.dialogs.quick_open.commands.reload",
+          "domain",
+          domainToName(this.hass.localize, domain)
+        ),
+      domain,
+      onClick: this._executeCommand,
```

Comment:
Add the actual service call here, so it stays flexible.

Reply:
Sorry, not sure I follow. Do you mean move the contents of `_executeCommand` directly into the onClick, like this?

```
private _generateReloadCommandItems(): CommandItem[] {
    const reloadableDomains = componentsWithService(this.hass, "reload").sort();
    return reloadableDomains.map((domain) => ({
      text: //...
      domain,
      onClick: () => {
        const eventData = { ... };
        await this.hass.callService( ... );
        fireEvent(this, "hass-service-called", eventData);
        this.closeDialog();
      }
   };
}
```

Reply:
No, just the params for `callService`

Reply:
So this?

```
private _generateReloadCommandItems(): CommandItem[] {
    const reloadableDomains = componentsWithService(this.hass, "reload").sort();

    return reloadableDomains.map((domain) => ({
      text: //...
      domain,
      onClick: () => this.hass.callService(domain, "reload")
   };
}
```

If that's what you're suggesting, then we'd lose the "hass-service-called" firing and the dialog wouldn't close after that's all done.

I could re-introduce those behaviors by making a callback that wraps onClick, and sending _that_ to the mwc-list-item's instead of the onClick directly:

```
<mwc-list-item
  @click=${this._wrappedServiceCall(onClick)}
  .domain=${domain}
  //...
>
```

Reply:
@bramkragten At the moment, this is the only unresolved review comment.

Are you okay with leaving it as `onClick: this._executeCommand` (which wraps the service call and the dialog close call in one callback)

Reply:
Added some suggestions, that should be ok for now

Reply:
Thank you for the commit suggestions. Changes pushed